server/routes.go

```diff
@@ -71,174 +73,143 @@ func modelOptions(model *Model, requestOpts map[string]interface{}) (api.Options
 	return opts, nil
 }
 
-func isSupportedImageType(image []byte) bool {
-	contentType := http.DetectContentType(image)
-	allowedTypes := []string{"image/jpeg", "image/jpg", "image/png"}
-	return slices.Contains(allowedTypes, contentType)
-}
-
-func (s *Server) GenerateHandler(c *gin.Context) {
-	checkpointStart := time.Now()
-	var req api.GenerateRequest
-	err := c.ShouldBindJSON(&req)
-
-	switch {
-	case errors.Is(err, io.EOF):
-		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "missing request body"})
-		return
-	case err != nil:
-		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
-		return
+func (s *Server) scheduleRunner(ctx context.Context, name string, caps []Capability, requestOpts map[string]any, keepAlive *api.Duration) (*runnerRef, error) {
```

Comment:
Passing in `caps` is weird here – shouldn't the caller do that before scheduling the runner?

Reply:
the caller doesn't have access to the model so it can't check capabilities

Reply:
It did before this change

Could we load the model and check the capabilities before deciding to schedule the model here? and then pass in the `Model`? It seems like a precondition the handler should check (ideally this function goes away and we do something like `schduler.Schedule(model, opts)` from the handler which would be a lot clearer

Reply:
maybe the function is misnamed. this doesn't schedule anything until some check including capabilities. it aggregates the various model and options related lines that chat, generate, and embeddings all have to call before scheduling a runner